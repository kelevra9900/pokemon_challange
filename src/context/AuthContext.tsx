import React, {createContext, useEffect, useReducer} from 'react';
// import AsyncStorage from '@react-native-async-storage/async-storage';
import {User, RegisterData, LoginData} from '../interfaces';
import {authReducer, AuthState} from './AuthReducer';

type AuthContextProps = {
  errorMessage: string;
  token: string | null;
  user: User | null;
  status: 'checking' | 'authenticated' | 'not-authenticated';
  signUp: (registerData: RegisterData) => void;
  signIn: (loginData: LoginData) => void;
  logOut: () => void;
  removeError: () => void;
};

const authInicialState: AuthState = {
  status: 'checking',
  token: null,
  user: null,
  errorMessage: '',
};

export const AuthContext = createContext({} as AuthContextProps);

export const AuthProvider = ({children}: any) => {
  const [state, dispatch] = useReducer(authReducer, authInicialState);

  useEffect(() => {
    checkToken();
  }, []);

  const checkToken = async () => {
    // No token, no autenticado
    // await AsyncStorage.setItem('token', JSON.stringify('token'));
    // if (!token) return dispatch({type: 'notAuthenticated'});
    // Hay token
    // const resp = await cafeApi.get('/auth');
    // if (resp.status !== 200) {
    //   return dispatch({type: 'notAuthenticated'});
    // }
    // await AsyncStorage.setItem('token', resp.data.token);
    // dispatch({
    //   type: 'signUp',
    //   payload: {
    //     token: resp.data.token,
    //     user: resp.data.usuario,,
    //   },
    // });
  };

  const signIn = async ({email, password}: LoginData) => {
    console.log('signIn', email, password);
    // try {
    //         const { data } = await cafeApi.post<LoginResponse>('/auth/login', { correo, password } );
    //   });
    //   dispatch({
    //     type: 'signUp',
    //     payload: {
    //       token: data.token,
    //       user: data.usuario,,
    //     },,
    //   });
    //   await AsyncStorage.setItem('token', data.token);
    // } catch (error) {
    //   dispatch({
    //     type: 'addError',
    //     payload: error.response.data.msg || 'Información incorrecta',,
    //   });
    // }
  };

  const signUp = async ({name, email, password}: RegisterData) => {
    console.log('signUp', name, email, password);
    // try {
    //         const { data } = await cafeApi.post<LoginResponse>('/usuarios', { correo, password, nombre } );
    //   });
    //   dispatch({
    //     type: 'signUp',
    //     payload: {
    //       token: data.token,
    //       user: data.usuario,,
    //     },,
    //   });
    //   await AsyncStorage.setItem('token', data.token);
    // } catch (error) {
    //   dispatch({
    //     type: 'addError',
    //     payload: error.response.data.errors[0].msg || 'Revise la información',,
    //   });
    // }
  };

  const logOut = async () => {
    // await AsyncStorage.removeItem('token');
    dispatch({type: 'logout'});
  };

  const removeError = () => {
    dispatch({type: 'removeError'});
  };

  return (
    <AuthContext.Provider
      value={{
        ...state,
        signUp,
        signIn,
        logOut,
        removeError,
      }}>
      {children}
    </AuthContext.Provider>
  );
};
