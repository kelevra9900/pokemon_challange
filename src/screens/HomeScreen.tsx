/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {View, Dimensions, Platform, FlatList} from 'react-native';

import {useSafeAreaInsets} from 'react-native-safe-area-context';

import {PokemonCard} from '../components/PokemonCard';
import {usePokemons} from '../hooks/usePokemons';
import {SearchInput} from '../components/SearchInput';
import {Loading} from '../components/ui/Loading';

const screenWidth = Dimensions.get('window').width;

export const HomeScreen = () => {
  const {simplePokemonList, loadPokemons} = usePokemons();
  const {top} = useSafeAreaInsets();

  return (
    <View
      style={{
        flex: 1,
        backgroundColor: '#111111',
      }}>
      <SearchInput
        style={{
          width: screenWidth - 51,
          top: Platform.OS === 'ios' ? top : top + 30,
        }}
      />
      <FlatList
        style={{
          top: 100,
          marginHorizontal: 10,
        }}
        data={simplePokemonList}
        keyExtractor={pokemon => pokemon?.id?.toString()}
        showsVerticalScrollIndicator={false}
        numColumns={2}
        renderItem={({item}) => <PokemonCard pokemon={item} />}
        onEndReached={loadPokemons}
        onEndReachedThreshold={0.4}
        ListFooterComponent={<Loading />}
      />
    </View>
  );
};
