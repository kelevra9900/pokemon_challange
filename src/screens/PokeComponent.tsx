import { StyleSheet } from "react-native";

export const themePokeComponent = StyleSheet.create({
    container: {
      marginHorizontal: 20,
      marginVertical: 7,
    },
    weightHeight: {
      marginTop: 50,
      height: 56,
      flexDirection: 'row',
      justifyContent: 'space-around',
    },
    contentEvo: {
      flexDirection: 'row',
      justifyContent: 'space-around',
      alignItems: 'center',
      marginHorizontal: 20,
      borderRadius: 8,
      height: 130,
      backgroundColor: 'rgba(44, 44, 44, 0.5)',
      borderColor: '#2c2c2c',
      borderWidth: 1,
    },
    title: {
      fontSize: 18,
      fontWeight: '600',
      color: 'white',
      marginTop: 20,
    },
    regularText: {
      fontSize: 16,
      color: 'white',
      fontWeight: '400',
    },
    subtitle: {
      fontSize: 12,
      color: '#828282',
      textAlign: 'center',
    },
    basicSprite: {
      width: 100,
      height: 65,
    },
    evoName: {
      fontSize: 14,
      fontWeight: '400',
      textAlign: 'center',
      color: 'white',
    },
  });
  