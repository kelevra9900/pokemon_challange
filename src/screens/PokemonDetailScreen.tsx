/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  Image,
  SafeAreaView,
  ActivityIndicator,
} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {StackScreenProps} from '@react-navigation/stack';

import {usePokemon} from '../hooks/useSinglePokemon';
import {FadeInImage} from '../components/ui/FadeImage';
import {PokemonDetails} from '../components/PokemonDetail';
import COLORS from '../utils/colors';
import {Loading} from '../components/ui/Loading';
import {ColorsType} from '../utils/color_types';

interface Props extends StackScreenProps<any, 'Pokemon'> {}

export const PokemonDetailScreen = ({route}: Props) => {
  const {simplePokemon, color} = route.params as any;
  const {id, name} = simplePokemon;
  const {top} = useSafeAreaInsets();

  const {isLoading, pokemon, details, evolutions} = usePokemon(id.toString());
  return (
    <SafeAreaView style={{flex: 1, backgroundColor: 'black'}}>
      {/* Head Containerr */}
      <View
        style={{
          backgroundColor: 'black',
          zIndex: 1,
        }}>
        <View style={styles.pokemonContent}>
          {/* Image in background */}
          <Image
            resizeMode="cover"
            source={{uri: simplePokemon.image}}
            blurRadius={25}
            style={styles.imageBlur}
          />
          {/* Pokemon Image */}
          <FadeInImage uri={simplePokemon.image} style={styles.pokemonImage} />
          {/* Pokemon Name */}
          <Text
            style={{
              ...styles.pokemonName,
              top: top - 7,
            }}>
            {name}
          </Text>
        </View>
        {/* Chips type */}
        {isLoading ? (
          <View style={styles.loadingIndicator}>
            <ActivityIndicator color={color} size={50} />
          </View>
        ) : (
          <>
            {/* Pokemon ID */}
            <Text style={styles.idPokemon}>{'#' + pokemon.id}</Text>
            <View style={styles.chips}>
              {pokemon.types.map(({type}) => (
                <View
                  key={type.name}
                  style={{
                    ...styles.containChip,
                    backgroundColor: ColorsType[type.name] || '#fff',
                    borderColor: ColorsType[type.name] || '#fff',
                  }}>
                  <Text style={styles.type}>{type.name}</Text>
                </View>
              ))}
            </View>
          </>
        )}
        {/* Pokemon Description */}
        <View style={styles.description}>
          <Text style={styles.textDescription}>{details}</Text>
        </View>
      </View>

      {/* Detalles y Loading */}
      {isLoading ? (
        <Loading />
      ) : (
        <PokemonDetails pokemon={pokemon} evolutions={evolutions} />
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  pokemonContent: {
    alignItems: 'center',
  },
  idPokemon: {
    color: '#fff',
    fontSize: 16,
    fontWeight: '400',
    textAlign: 'center',
  },
  imageBlur: {
    width: 480,
    height: 480,
    position: 'absolute',
    opacity: 0.5,
    top: -250,
    left: -40,
  },
  backButton: {
    position: 'absolute',
    zIndex: 1,
    left: 20,
  },
  pokemonName: {
    color: 'white',
    fontSize: 32,
    alignSelf: 'center',
  },
  pokemonImage: {
    width: 170,
    height: 170,
  },
  loadingIndicator: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  description: {
    marginVertical: 20,
    alignItems: 'center',
  },
  textDescription: {
    justifyContent: 'center',
    color: COLORS.greyLight,
  },
  regularText: {
    fontSize: 16,
    color: 'white',
    fontWeight: '400',
  },
  chips: {
    marginVertical: 6,
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'center',
  },
  containChip: {
    marginRight: 10,
    paddingHorizontal: 10,
    borderRadius: 95,
    width: 71,
    height: 24,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: 'red',
    borderWidth: 1,
  },
  type: {
    color: 'white',
    fontSize: 12,
  },
});
