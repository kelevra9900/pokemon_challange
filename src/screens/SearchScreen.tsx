/* eslint-disable react-native/no-inline-styles */
/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect, useState} from 'react';
import {
  View,
  Text,
  FlatList,
  Platform,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {ListPokemon} from '../components/ListPokemon';
import {InputComponent} from '../components/ui/Input';

import {Loading} from '../components/ui/Loading';
import {usePokemonSearch} from '../hooks/usePokemonSearch';
import {Pokemon} from '../interfaces/index';
import SearchIcon from '../assets/Icon/Search.svg';

export const SearchScreen = () => {
  const {top} = useSafeAreaInsets();
  const {isFetching, simplePokemonList} = usePokemonSearch();

  const [pokemonFiltered, setPokemonFiltered] = useState<Pokemon[]>([]);

  const [term, setTerm] = useState('');

  useEffect(() => {
    if (term.length === 0) {
      return setPokemonFiltered([]);
    }

    console.log(isNaN(Number(term)));

    if (isNaN(Number(term))) {
      setPokemonFiltered(
        simplePokemonList.filter(poke =>
          poke.name.toLocaleLowerCase().includes(term.toLocaleLowerCase()),
        ),
      );
    } else {
      const pokemonById = simplePokemonList.find(
        (poke: Pokemon) => poke.id.toString() === term,
      );
      setPokemonFiltered(pokemonById ? [pokemonById] : []);
    }
  }, [term]);

  if (isFetching) {
    return <Loading />;
  }
  const onChangeSearch = (query: string) => setTerm(query);

  return (
    <View style={styles.main}>
      <View style={styles.searchInput}>
        <InputComponent
          placeholder="Search"
          placeholderTextColor="#828282"
          styles={{width: term === '' ? '100%' : '84%'}}
          autoCorrect={false}
          value={term}
          onChangeText={onChangeSearch}
          icon={<SearchIcon width={22} height={22} />}
        />
        {term === '' ? null : (
          <TouchableOpacity onPress={() => setTerm('')}>
            <Text style={styles.cancel}>Cancel</Text>
          </TouchableOpacity>
        )}
      </View>

      {term === '' ? (
        <Text>Search pokemons by name or by number</Text>
      ) : (
        <FlatList
          data={pokemonFiltered}
          keyExtractor={pokemon => pokemon.id.toString()}
          showsVerticalScrollIndicator={false}
          // Header
          ListHeaderComponent={
            <Text
              style={{
                ...styles.title,
                ...styles.globalMargin,
                paddingBottom: 10,
                marginTop: Platform.OS === 'ios' ? top + 60 : top + 80,
              }}>
              {term}
            </Text>
          }
          renderItem={({item}: any) => <ListPokemon pokemon={item} />}
        />
      )}
    </View>
  );
};

export const styles = StyleSheet.create({
  main: {
    flex: 1,
    backgroundColor: '#111111',
    paddingHorizontal: 16,
  },
  cancel: {
    color: '#fff',
    fontSize: 14,
    marginTop: 10,
  },
  searchInput: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    zIndex: 1,
  },
  globalMargin: {
    marginHorizontal: 20,
  },
  title: {
    fontSize: 35,
    fontWeight: 'bold',
  },
});
