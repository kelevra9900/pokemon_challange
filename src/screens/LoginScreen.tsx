/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable react-native/no-inline-styles */
import React, {useContext, useEffect} from 'react';
import {
  View,
  Platform,
  KeyboardAvoidingView,
  Text,
  TouchableOpacity,
  SafeAreaView,
  Keyboard,
  Alert,
} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import {useNavigation} from '@react-navigation/native';

import {AuthContext} from '../context/AuthContext';

import LockIcon from '../assets/Icon/Lock.svg';
import EmailIcon from '../assets/Icon/Email.svg';
import WhiteLogo from '../components/WhiteLogo';
import {Background} from '../components/Background';
import {loginStyles} from '../theme/loginTheme';
import {Button} from '../components/ui/Button';
import {InputComponent} from '../components/ui/Input';
import {useForm} from '../hooks/useForm';

type mainScreenProp = StackNavigationProp<any, 'Main'>;

export const LoginScreen = () => {
  const {signIn, errorMessage, removeError} = useContext(AuthContext);

  const {email, password, onChange} = useForm({
    email: '',
    password: '',
  });

  const navigation = useNavigation<mainScreenProp>();

  useEffect(() => {
    if (errorMessage?.length === 0) {
      return;
    }

    Alert.alert('Login incorrecto', errorMessage, [
      {
        text: 'Ok',
        onPress: removeError,
      },
    ]);
  }, [errorMessage]);

  const handleLogin = () => {
    Keyboard.dismiss();
    signIn({email, password});
  };

  return (
    <>
      <Background />

      <KeyboardAvoidingView
        style={{flex: 1}}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
        <SafeAreaView style={loginStyles.formContainer}>
          {/* Logo */}
          <WhiteLogo />
          {/* Form View */}
          <View style={loginStyles.content}>
            {/* Input Email */}
            <InputComponent
              icon={<EmailIcon width={22} height={22} />}
              placeholder="Email"
              placeholderTextColor="#828282"
              onChangeText={(value: string) => onChange(value, 'email')}
              value={email}
              onSubmitEditing={handleLogin}
              autoCorrect={false}
              testID="email-input"
            />

            {/* Input Password */}
            <InputComponent
              placeholder="Password"
              placeholderTextColor="#828282"
              secureTextEntry={true}
              autoCorrect={false}
              onChangeText={(value: string) => onChange(value, 'password')}
              value={password}
              onSubmitEditing={handleLogin}
              icon={<LockIcon width={22} height={22} />}
              testID="password-input"
            />

            {/* login button */}
            <View style={loginStyles.buttonContainer}>
              <TouchableOpacity
                activeOpacity={0.8}
                style={loginStyles.button}
                testID="login-button"
                onPress={handleLogin}>
                <Text style={loginStyles.buttonText}>Log In</Text>
              </TouchableOpacity>
            </View>

            {/* Crear una nueva cuenta */}
            <Button
              testID="navigate-to-register"
              onPress={() => navigation.push('Register')}
            />
          </View>
        </SafeAreaView>
      </KeyboardAvoidingView>
    </>
  );
};
