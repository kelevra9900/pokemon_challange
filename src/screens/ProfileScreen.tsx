import React, {useContext} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {AuthContext} from '../context/AuthContext';
import {Background} from '../components/Background';

export const ProfileScreen = () => {
  const {user} = useContext(AuthContext);
  console.log('USUARIO', user);
  return (
    <>
      <Background />
      <View style={styles.content}>
        <View>
          <TouchableOpacity>
            <Text>Nombre de usuario</Text>
          </TouchableOpacity>
        </View>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  content: {
    alignItems: 'center',
    backgroundColor: '#111111',
  },
  cardContainer: {
    flex: 1,
  },
  container: {
    flex: 1,
  },
  headerContainer: {
    alignItems: 'center',
    marginBottom: 10,
    marginTop: 45,
  },
});
