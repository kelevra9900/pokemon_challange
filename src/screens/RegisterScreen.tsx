/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect, useContext, useState} from 'react';
import {
  View,
  Platform,
  KeyboardAvoidingView,
  Text,
  TouchableOpacity,
  Alert,
  Keyboard,
} from 'react-native';

import WhiteLogo from '../components/WhiteLogo';
import {loginStyles} from '../theme/loginTheme';
import {InputComponent} from '../components/ui/Input';
import {useForm} from '../hooks/useForm';
import {AuthContext} from '../context/AuthContext';

export const RegisterScreen = () => {
  const {signUp, errorMessage, removeError} = useContext(AuthContext);
  const [isKeyboardVisible, setKeyboardVisible] = useState(false);

  const {name, email, password, password_confirmation, onChange} = useForm({
    name: '',
    email: '',
    password: '',
    password_confirmation: '',
  });

  useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      () => {
        setKeyboardVisible(true); // or some other action
      },
    );
    const keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      () => {
        setKeyboardVisible(false); // or some other action
      },
    );

    return () => {
      keyboardDidHideListener.remove();
      keyboardDidShowListener.remove();
    };
  }, []);
  useEffect(() => {
    if (errorMessage?.length === 0) {
      return;
    }
    Alert.alert('Registro incorrecto', errorMessage, [
      {
        text: 'Ok',
        onPress: removeError,
      },
    ]);
  }, [errorMessage]);

  const onRegister = () => {
    Keyboard.dismiss();
    signUp({
      name,
      email,
      password,
      password_confirmation,
    });
  };

  return (
    <View style={loginStyles.main}>
      <KeyboardAvoidingView
        style={loginStyles.formContainer}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
        {/* Keyboard avoid view */}
        {isKeyboardVisible ? null : <WhiteLogo />}
        <View style={loginStyles.content}>
          {/* Name */}
          <InputComponent
            value={name}
            placeholder="Name"
            placeholderTextColor="#828282"
            onChangeText={(value: string) => onChange(value, 'name')}
            autoCorrect={false}
          />
          {/* Input Email */}
          <InputComponent
            value={email}
            placeholder="Email"
            placeholderTextColor="#828282"
            onChangeText={(value: string) => onChange(value, 'email')}
            autoCorrect={false}
          />

          {/* Input Password */}
          <InputComponent
            value={password}
            placeholder="Password"
            secureTextEntry={true}
            placeholderTextColor="#828282"
            onChangeText={(value: string) => onChange(value, 'password')}
            autoCorrect={false}
          />

          {/* Input Confirm Password */}
          <InputComponent
            value={password_confirmation}
            placeholder="Confirm Password"
            secureTextEntry={true}
            placeholderTextColor="#828282"
            onChangeText={(value: string) =>
              onChange(value, 'password_confirmation')
            }
            autoCorrect={false}
          />

          {/* Boton login */}
          <View style={loginStyles.buttonContainer}>
            <TouchableOpacity
              activeOpacity={0.8}
              style={loginStyles.button}
              onPress={onRegister}>
              <Text style={loginStyles.buttonText}>Create account</Text>
            </TouchableOpacity>
          </View>
        </View>
      </KeyboardAvoidingView>
    </View>
  );
};
