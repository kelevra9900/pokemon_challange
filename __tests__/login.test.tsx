import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import {LoginScreen} from '../src/screens/LoginScreen';
import {AuthProvider} from '../src/context/AuthContext';

describe('Should render Login page', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <LoginScreen />
      </NavigationContainer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('The login page must have a form with the following fields: email, password and a submit button', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = render(
      <NavigationContainer>
        <AuthProvider>
          <LoginScreen />
        </AuthProvider>
      </NavigationContainer>,
    );
  });
  it('Should contains Logo image', () => {
    expect(wrapper.getByTestId('logo-image')).toBeTruthy();
  });
  it('Should contains inputs (emial and password)', () => {
    waitFor(() => {
      expect(wrapper.getByTestId('email-input')).toBeTruthy();
      expect(wrapper.getByTestId('password-input')).toBeTruthy();
    });
  });
  it('page contains the login button', async () => {
    waitFor(() => {
      expect(wrapper.getByText('Log In')).toBeTruthy();
    });
  });
  it('page contains the register button', async () => {
    waitFor(() => {
      expect(wrapper.getByText('Create an account')).toBeTruthy();
    });
  });
  it('display required messages as the format: "The [field name] is required"', () => {
    const {getByTestId} = wrapper;
    fireEvent.changeText(getByTestId('email-input'), '');
    fireEvent.changeText(getByTestId('password-input'), '');
    fireEvent.press(getByTestId('login-button'));
  });
});

describe.only('The Login page must be define register button and navigate to this page', () => {
  it('Clicking register button', () => {
    const push = jest.fn();
    const {getByTestId} = render(
      <NavigationContainer>
        <LoginScreen />
      </NavigationContainer>,
    );
    fireEvent.press(getByTestId('navigate-to-register'));
    waitFor(() => {
      expect(push).toHaveBeenCalled();
    });
  });
});
