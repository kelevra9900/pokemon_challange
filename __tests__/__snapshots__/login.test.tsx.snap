// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  testID="login-page"
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "flex": 1,
          "height": 600,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "paddingHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 12,
          }
        }
      >
        <SvgMock
          height={38}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginTop": 12,
            }
          }
          testID="logo-image"
          width={204}
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F3F1F3",
              "borderRadius": 6,
              "elevation": 5,
              "flexDirection": "row",
              "height": 40,
              "marginTop": 20,
              "paddingHorizontal": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <SvgMock
            height={22}
            width={22}
          />
          <TextInput
            autoCorrect={false}
            icon={
              <SvgMock
                height={22}
                width={22}
              />
            }
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            placeholderTextColor="#828282"
            style={
              Object {
                "color": "#000",
                "marginLeft": 5,
              }
            }
            testID="email-input"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F3F1F3",
              "borderRadius": 6,
              "elevation": 5,
              "flexDirection": "row",
              "height": 40,
              "marginTop": 20,
              "paddingHorizontal": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <SvgMock
            height={22}
            width={22}
          />
          <TextInput
            autoCorrect={false}
            icon={
              <SvgMock
                height={22}
                width={22}
              />
            }
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            placeholderTextColor="#828282"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "marginLeft": 5,
              }
            }
            testID="password-input"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 30,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2F80ED",
                "borderRadius": 8,
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 5,
                "width": "100%",
              }
            }
            testID="login-button"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                }
              }
            >
              Log In
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 30,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="navigate-to-register"
          >
            <Text
              style={
                Object {
                  "color": "#2F80ED",
                  "fontSize": 16,
                  "textDecorationLine": "underline",
                }
              }
            >
              Create an account
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
